### ISSUE TITLE
pagination not working on Home page

### ISSUE THREAD

**Issue by Vishakha-Patel** [Codeinwp/themeisle-companion#267]
Ticket: https://secure.helpscout.net/conversation/612085274/164552?folderId=949810

Post Type Grid widget's pagination does not work on Home page. It changes the URL but actual posts remain same.

@selul

**Comment by selul** [comment]
@rodica-andronache maybe one of the boys can help.

**Comment by rodica-andronache** [comment]
@contactashish13 would please take a look at this?
It also reported here https://wordpress.org/support/topic/post-grid-pagination-issue-2/

**Comment by contactashish13** [comment]
@Vishakha-Patel @rodica-andronache what options do I need to choose (and what theme to use) to reproduce this?

**Comment by Vishakha-Patel** [comment]
@contactashish13 You can use any theme. Try with Hestia.

Edit its Homepage with Elementor plugin. Add "Post Type Grid" widget and turn on "Pagination" option.

When you'll check this, posts on the second page won't come.

For example, see the pagination on this website - http://natoriousb.com/ from the ticket.

**Comment by contactashish13** [comment]
@rodica-andronache PR: https://github.com/Codeinwp/elementor-extra-widgets/pull/9

**Comment by rodica-andronache** [comment]
@contactashish13 from what I've tested, with this change, the pagination is working on the frontpage, but when used on an internal page, is not. Could you please take a look?

**Comment by contactashish13** [comment]
@rodica-andronache that was indeed a problem. have fixed it. PR: https://github.com/Codeinwp/elementor-extra-widgets/pull/10

**Comment by rodica-andronache** [comment]
@contactashish13 thanks 👍 it indeed works on all pages now. But on posts is still not working

**Comment by contactashish13** [comment]
@rodica-andronache Its working for me. Can you list down the exact steps to reproduce this problem?

**Comment by rodica-andronache** [comment]
@contactashish13 just install OBFX (latest version) , create a **post** and in that post, add the post grid ( with the post type : post ) , and the navigation is not working 
https://prnt.sc/k5nxhm

**Comment by contactashish13** [comment]
@rodica-andronache I tried the steps with post type post as well as a custom post type and it seems to be working fine. A post is an internal page so there is no reason for it not to work. 

Can you check if custom post types work fine?

**Comment by rodica-andronache** [comment]
@contactashish13 I checked with a custom post type too. Still not working. Are you sure you've tested on a post, not a page? 
I'm asking because I think that's the reasons is not working, because it's interfering with the posts loop

**Comment by rodica-andronache** [comment]
@contactashish13 check this http://pleasant-echidna.w5.poopy.life/2018/07/30/elementor-26/ 
user: ash, password: ash

**Comment by contactashish13** [comment]
@rodica-andronache this instance expired. can you please share another one?

**Comment by rodica-andronache** [comment]
@contactashish13 http://hilarious-markhor.w5.poopy.life/2018/08/02/elementor-1705/
user: ash,
pass: ash

**Comment by contactashish13** [comment]
@rodica-andronache I see the problem and its very strange that I cannot replicate it on my dev instance with the same version of Elementor.

**Comment by rodica-andronache** [comment]
yes, it's weird it's working for you. Be sure to have also the latest version of OBFX

**Comment by rodica-andronache** [comment]
ok, you can try on a different instance maybe? It's clear there is a problem :D
